### PR TITLE
nextpnr: restructure a bit and add pkg images

### PR DESCRIPTION
**Pull request by se-bi**
Hey,
next one, also as discussed in #4.

(builds on #13)

### CODE REVIEW

File: .github/workflows/nextpnr.yml [hunk 1]
```diff
@@ -43,16 +44,24 @@ jobs:
...
     - run: echo "$(pwd)/.github/bin" >> $GITHUB_PATH
 
-    - run: dockerBuild nextpnr:ice40      nextpnr ice40
-    - run: dockerBuild nextpnr:icestorm   nextpnr icestorm
-    - run: dockerBuild nextpnr:ecp5       nextpnr ecp5
-    - run: dockerBuild nextpnr:prjtrellis nextpnr prjtrellis
-    - run: dockerBuild nextpnr            nextpnr
+    - run: dockerBuild nextpnr:ice40       nextpnr ice40
+    - run: dockerBuild nextpnr:icestorm    nextpnr icestorm
+    - run: dockerBuild pkg:nextpnr-ice40   nextpnr pkg-ice40
```
Comment (eine) [inline .github/workflows/nextpnr.yml:49]: Please, move `pkg` images first:

- pkg:nextpnr-ice40
- nextpnr:ice40
- nextpnr:icestorm
- pkg:nextpnr-ecp5
- ...

From smallest to largest. 

I would try keeping the same order for testing and pushing.

**Review by eine — CHANGES_REQUESTED** [review]
Please, address the comment below and rebase on top of `main`. This seems to contain some modifcations from the other PR (already merged).

**Comment by eine** [comment]
LGTM. Do you want to handle doc modifications in this same PR, or will be finish the technical modifications first and we will later document all the changes at once?

**Comment by se-bi** [comment]
> LGTM. Do you want to handle doc modifications in this same PR, or will be finish the technical modifications first and we will later document all the changes at once?

valid point! Though first reaction was kind of :see_no_evil: 

After a quick look, seems a lot generated stuff, is it mainly based on the `doc/tools.yml`?

**Comment by eine** [comment]
> After a quick look, seems a lot generated stuff, is it mainly based on the `doc/tools.yml`?

Yes, in this case it's editing `doc/tools.yml` and that will update the table. Then, the graph should be updated. That's trickier because the `dot` source needs to be edited manually.

Let's merge this as is. In the next PR, please update `tools.yml`; and don't worry about the graph. I'll see how to handle that.

**Comment by se-bi** [comment]
what else, than:
```
 nextpnr:
   url: 'https://hdl.github.io/awesome/items/nextpnr'
+  pkg:
+    - 'nextpnr-ice40'
+    - 'nextpnr-ecp5'
+    - 'nextpnr-generic'
   use:
     - 'nextpnr'
     - 'nextpnr:ice40'
     - 'nextpnr:ecp5'
+    - 'nextpnr:generic'
```

**Comment by eine** [comment]
Nothing actually! I believe that's enough for this PR! I was mislead by the next one, which will imply modifying `impl`. Please, push that here.

**Comment by se-bi** [comment]
Ok, also done.

...`impl` for another time.

**Pull request merged**